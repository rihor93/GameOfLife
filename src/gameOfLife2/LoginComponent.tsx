import React, { useState } from "react";


export interface LoginComponentProps {
    onClick: (name: string) => void,
}

const LoginComponent: React.FC<LoginComponentProps> = (props) => {

    const [name, setName] = useState('');

    return (

        <div className="cardBorder contentWrapper">
            <div>Введите имя:
                <input data-testid="loginInput" value={name} onChange={(e: React.ChangeEvent<HTMLInputElement>) => { setName(e.target.value) }}></input>
            </div>
            <div>
                <button data-testid="btnStart" onClick={() => props.onClick(name)}>Старт!</button>
            </div>
        </div>

    );
}



export default LoginComponent;