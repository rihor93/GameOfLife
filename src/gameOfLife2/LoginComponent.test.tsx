import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import LoginComponent from './LoginComponent';

describe('LoginComponent tests', () => {
    it('show id on screen', () => {

        render(<LoginComponent onClick={(name: string) => { console.log(name) }} />);
        const element = screen.getAllByTestId('loginInput');

        expect(element).toHaveLength(1);
    });



});

describe('LoginComponent SnapshotTest', () => {
    /*render(<LoginComponent onClick={(name: string) => { console.log(name) }} />);
    const element = screen.getAllByTestId('loginInput');
    expect(element).toHaveLength(1);
    fireEvent.change(element[0], {target: {value: 'user'}})*/

    it('snapshottest', () => {
        const cellContainer = renderer
            .create(<LoginComponent onClick={()=> {}} />)
            .toJSON();
        expect(cellContainer).toMatchSnapshot();
    });




});
