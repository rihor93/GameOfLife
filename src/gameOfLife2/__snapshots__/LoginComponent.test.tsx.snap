// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent SnapshotTest snapshottest 1`] = `
<div
  className="cardBorder contentWrapper"
>
  <div>
    Введите имя:
    <input
      data-testid="loginInput"
      onChange={[Function]}
      value=""
    />
  </div>
  <div>
    <button
      data-testid="btnStart"
      onClick={[Function]}
    >
      Старт!
    </button>
  </div>
</div>
`;
